src/gentropy/datasource/gwas_catalog/associations.py

```diff
@@ -195,7 +194,7 @@ def _collect_rsids(
         return f.array_distinct(f.array(snp_id, snp_id_current, risk_allele))
 
     @staticmethod
-    def _map_variants_to_variant_index(
+    def _map_variants_to_gnomad_variants(
```

Comment:
This method could be abstracted from the type of variant index you pass. As far as I can see, the main difference is the application of `convert_gnomad_position_to_ensembl`.

Do you think this method could live in the gnomad parser? In fact it'd make sense to convert them before dumping the gnomad variants dataset. When we generate the variant index, we join the VEP output with the Gnomad dataset to get the AF. Aren't we joining 2 datasets with 2 diff coordiate systems?
Out of scope of the PR, just a comment

Reply:
I don't have strong feeling about this. Currently we are only calling the `convert_gnomad_position_to_ensembl` method here. I quite forgot what was happening in the variant index branch. If we need to touch that code, it can be re-designed.